sw/device/lib/testing/json/command.h

```diff
@@ -20,6 +20,10 @@ extern "C" {
     value(_, I2cReadTransaction) \
     value(_, I2cWriteTransaction) \
     value(_, I2cWriteTransactionSlow) \
+    value(_, MemRead) \
```

Comment:
In term of reusability, it's not ideal to have all related commands flattened in this enumeration. This prevents having a `mem` C-library that dispatches them so you have to manually call the various ujson mem functions. One possibility would be to have a single `Mem` command which then has subcommands `Read`, `Read32` and so on. This way the `mem` library can have a single `ujson_mem` command, even if it's slightly less efficient.

Reply:
I'm not sure that approach is best here. There's no need to increase the amount of I/O stages for reusability: You can achieve it by using the same pattern as distributed argument handling.

```c
TRY(ujson_deserialize_test_command_t(uj, &command));
/* Common command dispatch */
status_t status = ottf_ujcmd_dispatch(uj, command);
switch (status) {
  kUnimplemented:
    /* Test-specific or other commands, if needed */
    TRY(handle_test_specific_command(uj, &command));
    break;
  kOk:
    break;
  default:
    return status;
}
```